### ISSUE TITLE
Results viewer: DELETE key incorrectly deletes rows

### ISSUE THREAD

**Issue by lemon-juice** [dbeaver/dbeaver#6393]
In my preferences the shortcut key to delete a row in the results pane is Alt+Delete (the default one). However, DBeaver also will delete a row when I press Delete without Alt. Here are the key preferences:

![dbeaver-delete-keys](https://user-images.githubusercontent.com/11840364/62240622-a6d79e80-b3d7-11e9-81a7-38842de4c736.png)

I don't know how long this behavior has been in DBeaver but I noticed it today when I wanted to add a new row by duplicating an existing row. It was an unpleasant experience because there were many columns in the table and I spent some time filling in all the values and I accidentally pressed Delete while not having the cursor focused in a cell (because I used the Delete key a lot while editing some of the column values). The not yet saved row was deleted and there seemed to be no way of restoring my new data and I had to start again. The default Alt+Delete key is a safe choice as it makes it hard to delete data and the standard Delete operation should not delete the row (it might for example make the selected cell value NULL but not delete the whole row).

A related bug that I've noticed here: First I accidentally pressed Delete when adding a new row in the Record view and the Delete key did not make the editing row disappear but only moved the cell selection to the top of the page so I didn't even know my data would be lost. Only when I pressed Save I noticed that my data were not saved. This is another bug - when I delete a new row in the Record view the data should disappear so that I can see the delete operation has been executed.

**Comment by serge-rider** [comment]
Delete shortcut was there for a years. A lot of people get used to it.
It is a very convenient shortcut in most cases.
So I don;t know how to solve this..

**Comment by p91paul** [comment]
There are 2 commands: 'Delete', working "In windows", and 'Delete current row', working in "Results viewer context". What if both are bound to the Del key by default, but the former *does not* delete current row (i.e. is ignored when on results viewer)?
Then @lemon-juice could remap 'Delete current row' to Alt+Del and will live happily with the Del key doing nothing on resultsets, while the rest of us continue using the Del key.

Don't know if this is feasible with Eclipse keybindings.

**Comment by lemon-juice** [comment]
Yes, I can see now how the pure Delete key is convenient for deleting rows, though I never used it that way. I first thought about a similar solution to what @p91paul suggests with separate key bindings - then I could simply remove the key binding for the Delete key. However, I think there is a better solution to solve the problem, which I think exists elsewhere.

The thing is that there is no problem with the pure Delete key used for deleting existing rows: the rows are marked for deletion with the red background and there's always an option to cancel the action if I've marked wrong rows. However, the problem is with new rows that have been added with the 'Add new row' or 'Duplicate current row' action. If I delete a new row then it is not marked for deletion, it disappears from the view immediately. And if I delete a new row in the Record view it is deleted although it does not disappear from the view - which is misleading and looks like a bug. In either case there is a (too) easy way of deleting new rows with the Delete key without any way to bring them back.

What I'm suggesting is:

1. When deleting new rows make the behaviour the same as when deleting existing rows: keep them in the view and mark them for deletion (or rather not-insertion) with the red background (also in the record view).

2. Add an action to revert the deletion (un-mark for deletion, remove the red background). Then if I press Delete by accident I can see immediately the row gets red background and I can revert it back. This revert action could get its own context menu entry and a shortcut key (like Ctrl+Delete or Ctrl+Z) or the same Delete action could be used for both deleting and reverting - either solution would be fine for me.

I think this would solve the real problem and additionally fix some weird and buggy behaviour. Additionally, we would get a possibility to revert rows from deletion even for the existing ones, which could come in handy sometimes.
